app/controllers/sog/name-trainer.js

```diff
@@ -0,0 +1,145 @@
+import Controller from '@ember/controller';
+import { computed } from '@ember/object';
+import GroupMembershipsMixin from 'alpha-amber/mixins/group-memberships-mixin';
+import { run } from '@ember/runloop';
+
+export default Controller.extend(GroupMembershipsMixin, {
+  queryParams: ['group', 'difficulty'],
+  group: null,
+  difficulty: 1,
+  selectedGroup: null,
+  difficultyOptions: [
+    {
+      value: 1,
+      label: 'Makkelijk'
+    }, {
+      value: 2,
+      label: 'Gemiddeld'
+    }, {
+      value: 3,
+      label: 'Moeilijk'
+    }
+  ],
+  current: 1,
+  success: 0,
+  questions: [],
+  textInput: '',
+  started: false,
+  finished: false,
+  answered: false,
+  groupSelected: computed('group', 'model', function() {
+    this.generateQuestions(this.get('model'));
+    return this.get('group') != null;
+  }),
+  humanizedDifficulty: computed('difficultyOptions', 'difficulty', function() {
+    return this.get('difficultyOptions').find(option => option.value === this.get('difficulty')).label;
+  }),
+  currentQuestion: computed('current', 'questions', function() {
+    return this.get('questions').objectAt(this.get('current') - 1).question;
+  }),
+  currentOptions: computed('current', 'questions', function() {
+    return this.get('questions').objectAt(this.get('current') - 1).options;
+  }),
+  currentQuestionItem: computed('current', 'questions', function() {
```

Comment:
Hebben we bovenstaande methods nodig? Kunnen we deze niet gewoon gebruiken en dan .options/.question hierop aanroepen? Misschien ook hernoemen naar currentQuestion

Reply:
Nee, niet echt inderdaad. Zijn verwijderd.